conan/tools/cmake/cmakedeps.py

```diff
@@ -615,6 +555,20 @@ def _data_filename(self, pkg_filename):
         data_fname += "-data.cmake"
         return data_fname
 
+    def get_name(self, req):
+        return req.cpp_info.get_property("cmake_target_name", self.name) or req.ref.name
```

Comment:
This is good, but we might have a transition problem, the CMakeDeps will not work with ConanCenter packages in a very long time. Maybe we should consider postponing this.

Reply:
done